kostyor/upgrades/strategy.py

```diff
@@ -0,0 +1,24 @@
+import abc
```

Comment:
I think the strategy.py must be a separate commit (and maybe even another PR), since it's not about "Decision engine: Call driver pre-upgrade hooks"